discovery-infra/tests/test_install.py

```diff
@@ -2,12 +2,28 @@
 from tests.base_test import BaseTest
 from tests.conftest import get_api_client
 
+from test_infra.junit_report import JunitTestCase, JunitTestSuite
+
 
 class TestInstall(BaseTest):
+
+    @JunitTestSuite("./reports")
     @pytest.mark.parametrize("openshift_version", list(get_api_client().get_openshift_versions().keys()))
     def test_install(self, nodes, cluster, openshift_version):
         # Define new cluster
-        new_cluster = cluster(openshift_version=openshift_version)
-        new_cluster.prepare_for_install(nodes=nodes)
-        # Install Cluster
+        new_cluster = self.get_cluster(cluster, openshift_version)
+        self.prepare_for_install(new_cluster, nodes)
+        # # Install Cluster
+        self.start_install_and_wait_for_installed(new_cluster)
+
+    @JunitTestCase()
+    def get_cluster(self, _cluster, _openshift_version):
+        return _cluster(openshift_version=_openshift_version)
+
+    @JunitTestCase()
+    def prepare_for_install(self, new_cluster, _nodes):
+        new_cluster.prepare_for_install(nodes=_nodes)
```

Comment:
It is possible to decorate the inner functions instead of creating a new function for each usage?
Then, it should look in case it is running inside a suite / if a suite is open

Reply:
Sure. That will work.
But in the current design, if you decorate inner function with `JunitTestCase` and it don't have parent suite it will throw an `SuiteNotExistError` exception, So it's better to use it only with `JunitTestSuite`.